### DISCUSSION TITLE
type file not being generated and no related tsserver logs exist

### DISCUSSION THREAD

**Discussion by aydangoon** [0no-co/gql.tada#88] [Q&A] [answered]
Hi, I'd really like to use gql.tada for my nextjs project but I'm running into issues generating the type file. I've followed the debug steps mentioned in https://github.com/0no-co/gql.tada/discussions/49#discussion-6167593. I've verified:
* my graphql server is definitely running
* I'm using my workspace's version of typescript (5.2.2)
* exploring the TSServer logs, i see 0 logs for `[GraphQLSP]`

here is my entire tsconfig file:

```json
{
  "compilerOptions": {
    "paths": {
      "~/*": ["./src/*"],
      "public/*": ["./public/*"]
    },
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "skipLibCheck": true,
    "strict": true,
    "noEmit": true,
    "incremental": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve"
  },
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules"],
  "plugins": [
    {
      "name": "@0no-co/graphqlsp",
      "schema": {
        "url": "<schema endpoint>",
        "headers": {
          "Authorization": "<api key>"
        }
      },
      "tadaOutputLocation": "./src/graphql.ts"
    }
  ]
}
```

Are there additional diagnostics steps i can take? pls i need me some typesafe graphql 😢

**Comment by kitten** [comment]
Even if your VSCode is loading the workspace TypeScript version, it kind of looks like you're in a monorepo. Maybe you've opened the workspace/project on a different level, so a different `tsconfig.json` without the plugin is being opened?

**Reply by aydangoon** [reply]
I'm in a single repo bootstrapped with `create-next-app`, Additionally when I click on open config file in the status bar
<img width="291" alt="Screenshot 2024-02-27 at 2 11 00 PM" src="https://github.com/0no-co/gql.tada/assets/57046728/9c4a30cc-0050-4769-995f-e3b684f0f8ce">

it does take me to this `tsconfig.json` file.

**Reply by JoviDeCroock** [reply, marked as answer]
And when you click select version it sais workspace right? Have you restarted the tsserver and enabled logs? Maybe there is a crash log somewhere

**Reply by aydangoon** [reply]
After reducing the tsserver log level from "verbose" to "normal", I was able to see in the logs that `GraphQLSP` was throwing a syntax error in parsing our schema because our schema had a syntax error (actually kms 💀). Generation is working now, thanks for the quick support!

As an aside, if possible it might be nice to have easier surfacing of errors? Having to jump into a logfile then from there to the graphqlsp/dist code to determine the issue was a bit of a headache haha

**Reply by JoviDeCroock** [reply]
Sadly ts does not offer a better way yet 😅
